src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -379,66 +289,39 @@ def forward(
         else:
             present = None
 
-        # [batch_size, head_dim, q_length, k_length]
-        output_size = (query_layer.size(0), query_layer.size(2), query_layer.size(1), key_layer.size(1))
-
-        # [batch_size, q_length, num_heads, head_dim] -> [q_length, batch_size * num_heads, head_dim]
-        query_layer = query_layer.transpose(1, 0).reshape(output_size[2], output_size[0] * output_size[1], -1)
-
-        # [batch_size, k_length, num_heads, head_dim] -> [k_length, batch_size * num_heads, head_dim]
-        key_layer = key_layer.transpose(1, 0).reshape(output_size[3], output_size[0] * output_size[1], -1)
-
-        # slice alibi tensor until the query length
-        sliced_alibi = alibi[: output_size[0] * output_size[1], :, : output_size[3]]
-
-        # Raw attention scores. [batch_size * num_heads, q_length, k_length]
         beta = 1.0 / self.layer_number
 
-        matmul_result = torch.baddbmm(
-            sliced_alibi,
-            query_layer.transpose(1, 0),
-            key_layer.transpose(1, 0).transpose(1, 2),
-            beta=beta,
-            alpha=(1.0 / self.norm_factor),
-        )
+        # # [batch_size*num_heads, head_dim, q_length] x [batch_size*num_heads, head_dim, k_length] -> [batch_size*num_heads, q_length, k_length]
+        matmul_result = (1.0 / self.norm_factor) * torch.bmm(
```

Comment:
Why was `baddbmm` dropped? Feels like exactly our usecase no?

Reply:
`baddbmm` was initially replaced by `bmm` after our several experiments to try and solve the differences noticed in using half precisions (`fp16` and `bf16`). Check our findings [here](https://github.com/younesbelkada/transformers/blob/bf311482284525423f9cf58f15cb006bcf52bb4d/tests/models/bloom/test_modeling_bloom.py#L379).
The second reason is that, we were also investigating the necessity of using alpha and beta coefficients. As they were [deemed unnecessary in the FLAX implementation](https://github.com/huggingface/transformers/commit/3444985451749540bd862910ebe7d01da8413771)

Reply:
edit: I linked the wrong findings. I meant to link the comments for the test `test_batch_generation_padd`
https://github.com/younesbelkada/transformers/blob/eb86c4369d465d6181a865e43fd198b28c0b7a02/tests/models/bloom/test_modeling_bloom.py#L450